#!/bin/bash

# This script will run BUSCO to assess genome assembly completeness

###############
###  SETUP  ###
###############

###  Conda  ###
echo "Setting up conda environment"
source /home/sejoslin/.bashrc
conda activate BUSCO
export AUGUSTUS_CONFIG_PATH="/home/sejoslin/config/augustus-3.2.3/config"

###  Modules ###
#echo "Loading modules."
#module load busco/2.0
#export AUGUSTUS_CONFIG_PATH="/home/sejoslin/config/augustus-3.2.3/config"
#BUSCO.py --version


###  Variables  ###
echo "Setting up variables"

THREADS=${SLURM_NTASKS}

# external variables
COVERAGE=$1		# Coverage level of assembly
PREFIX=$2		# Prefix of all fastq files (and output)
ASM=$3			# Letter corresponding to the assembly
ASSEMBLY=$4		# Supernova assembly file
OUT_DIR=$5		# Path to report directory
LINEAGE=$6		# Path to lineage

# internal variables
busco_dir="${OUT_DIR}/busco"

echo "  Coverage: ${COVERAGE}"
echo "  Prefix: ${PREFIX}"
echo "  Assembly Letter: ${ASM}"
echo "  Assembly Directory: ${ASSEMBLY}"
echo "  Output Diectory: ${OUT_DIR}"
echo "  Lineage File: ${LINEAGE}"
echo "  BUSCO Directory: ${busco_dir}"
echo "  Threads: ${THREADS}"
echo ""



###################
###  RUN BUSCO  ###
###################

mkdir -p ${busco_dir}
cd ${busco_dir}

BUSCO_call="run_BUSCO.py -i ${ASSEMBLY} \
	-l ${LINEAGE} \
	-o BUSCO_${PREFIX}_${COVERAGE}x${ASM} \
	-m genome \
	-c $((THREADS - 2))"
echo ${BUSCO_call}
eval ${BUSCO_call}



## BUSCO README ##
cat << readme >> README.md
This directory contains output for the reports generated by $(run_BUSCO.py --version) for the assembly of Hypomesus transpacificus.
Files & Directories:
    --short_summary_*.txt	== Contains a plain text summary of the results in BUSCO notation and give brief breakdown of metrics.
    --full_table_*.tsv		== Contains the complete results in a tabular format with scores and lengths of BUSCO matches, and coordinates.
    --missing_buscos_list_*.tsv	== Contains a list of missing BUSCOs.
    --hmmer_output		== Directory containing HMMER output of searches with BUSCO HMMs
    --blast_output 		== tBLASTn results
    --augustus_output		== Augustus-predicted genes
    --single_copy_busco_seq	== FASTA format file for each complete single-copy BUSCO identified.
readme



